public/calendar_reboot.js

```diff
@@ -50,35 +50,52 @@ function CheckIt() {
     }.bind(this));
    
    
-    //WHEN PAGE LOADS
-    //first load the allCalendarIds from storage.
-    var allCalendarIds = this.store.getAllCalendarIds() || {};
-   
-    //GOING THROUGH THE KEYS OF THE DICTIONARY allCalendarIds
-    for (var key in allCalendarIds) {
-      if (allCalendarIds.hasOwnProperty(key)) {
-        this.addCalendarToDropdown(key, allCalendarIds[key]);
-      }
-    }
+    // WHEN PAGE LOADS
+    // Load the calendar Ids from storage and fill the dropdown with calendar
+    // titles.
+    this.store.getAllCalendarIds()
+        .then(function (allCalendarIds) {
+            // Add calendar titles to dropdown.
+            
+            for (var key in allCalendarIds) {
+                if (allCalendarIds.hasOwnProperty(key)) {
+                this.addCalendarToDropdown(key, allCalendarIds[key]);
+                }
+            }
+        }.bind(this))
+        
+        .catch(function (value) {
+            console.log("No calendars in storage.");
+            return;
+        }.bind(this));
+    
+    
    
-    //get the current_active calendar id from storage if any
-    var activeCalendarId = this.store.getActive();
     
-    if (activeCalendarId !== null) {
-        
-        var activeCalendarState = this.store.loadById(activeCalendarId);
-        
-        if (activeCalendarState !== null) {
-            var state = activeCalendarState;
-            var calendar = new Calendar(state, this);
+    // Get the current active calendar from storage and display it.
+    // If there is none, show build calendar menu.
+   this.store.getActive()
+       .then(function (activeCalendarId) {
            
-            this.displayCalendar(calendar);
-        }
-        
-    }
-    else {
-        this.showBuildMenu();
-    }
+           this.store.loadById(activeCalendarId)
+               .then(function (activeCalendarState) {
+                   if (activeCalendarState !==  null) {
+                       var state = activeCalendarState;
+                       var calendar = new Calendar(state, this);
+                       this.displayCalendar(calendar);
+                   }
+               }.bind(this))
+               
+               .catch(function() {
+                   this.showBuildMenu();
```

Comment:
If we got here then that means that our `activeCalendar` doesn't exist. This is a serious error and means the data is corrupted somehow. You should at a minimum `console.error()` it and maybe even try to correct the data.